lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -984,8 +988,37 @@ def restart_all_services(self):
             )
         elif self._init_system == InitMode.DOCKER:
             self.exec_command("cd /home/vagrant/magma/lte/gateway/docker && docker-compose restart")
-        print("Waiting for all services to restart. Sleeping for 60 seconds..")
-        self.wait_for_restart_to_finish(60)
+
+        self._wait_for_pipelined_to_initialize()
+
+        EXTRA_WAIT_TIME_FOR_OTHER_SERVICES_SECONDS = 10
+        print(f"Waiting {EXTRA_WAIT_TIME_FOR_OTHER_SERVICES_SECONDS} seconds to ensure all services restarted ...")
+        time.sleep(EXTRA_WAIT_TIME_FOR_OTHER_SERVICES_SECONDS)
+
+    def _wait_for_pipelined_to_initialize(self):
+        """
+        Introduced, because pipelined is the first service the tests communicate with and
+        it has been observed that the previous static waiting time is not sufficient.
+        """
+        print("Waiting for pipelined to be started ...")
+        wait_time_seconds = 0
+
+        WAIT_INTERVAL_SECONDS = 5
+        MAX_WAIT_SECONDS = 120
+        print(f"  check every {WAIT_INTERVAL_SECONDS} seconds (max {MAX_WAIT_SECONDS} seconds) if pipelined is started ...")
+        pipelined_is_running, datapath_is_initialized = get_datapath_state()
+        while not datapath_is_initialized:
+            if pipelined_is_running:
+                print(f"  datapath not yet initialized for {wait_time_seconds} seconds ...")
+            else:
+                print(f"  pipelined not yet running for {wait_time_seconds} seconds ...")
+            time.sleep(WAIT_INTERVAL_SECONDS)
```

Comment:
After the last sleep we are not actually verifying the datapath state. e.g., check for `wait_time_seconds = 120`. So, last sleep of `WAIT_INTERVAL_SECONDS` is useless because after that last sleep we are only checking `if wait_time_seconds > MAX_WAIT_SECONDS:` in line number 1018 and returning failure with exception.

Logic can be similar to the code snippet in the other review comment.